crates/ra_hir/src/code_model_api.rs

```diff
@@ -213,8 +226,39 @@ impl Enum {
         Ok(db.enum_data(self.def_id)?.name.clone())
     }
 
-    pub fn variants(&self, db: &impl HirDatabase) -> Cancelable<Vec<(Name, Arc<VariantData>)>> {
-        Ok(db.enum_data(self.def_id)?.variants.clone())
+    pub fn variant_ids(&self) -> &[DefId] {
```

Comment:
this should return `EnumVariant`s (we know all the variants are enum variants, so we should return the typed wrapper). It'll probably have to return `impl Iterator<Item = EnumVariant>`, but that should work (I hope)

Reply:
Hmm actually after thinking about it, returning an Iterator of `EnumVariant`s will probably be quite hard and require writing it by hand since it would need to own the `Arc<EnumData>`... it's probably fine to collect the `EnumVariant`s into a `Vec` though

Reply:
returning a `Vec` now